geomstats/geometry/pre_shape.py

```diff
@@ -513,6 +847,38 @@ def curvature(
         curvature = gs.reshape(curvature, max_shape)
         return curvature
 
+    def curvature_derivative(
+            self, unused_hor_tg_vec_h, hor_tg_vec_x,
+            unused_hor_tg_vec_y, unused_hor_tg_vec_z,
+            unused_base_point=None):
+        r"""Compute the covariant derivative of the curvature.
+
+        For four horizontal tangent vectors at a base point :math: `H|_P, X|_P,
+        Y|_P, Z|_P` given in argument, the covariant derivative of the
+        quotient curvature :math: `(\nabla_H R)(X, Y)Z |_P` is computed at the
+        base point P. Since the sphere is a constant curvature space this
+        vanishes identically.
+
+        Parameters
+        ----------
+        hor_tg_vec_h : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        hor_tg_vec_x : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        hor_tg_vec_y : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        hor_tg_vec_z : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        base_point : array-like, shape=[..., n, n]
+            Point on the group.
+
+        Returns
+        -------
+        curvature_derivative : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        """
+        return gs.zeros_like(hor_tg_vec_x)
```

Comment:
The link with the sphere is in the docstring. I see this method as a more efficient implementation since it is avoid the call to the hypersphre function.